fixit/common/testing.py

```diff
@@ -116,6 +116,11 @@ def _test_method(
                     f"Expected:\n    {test_case.expected_str}\nBut found:\n    {report}"
                 )
 
+            if test_case.message is not None and test_case.message != report.message:
+                raise AssertionError(
+                    f"Expected message:\n    {test_case.message}\nBut got:\n    {report.message}"
+                )
+
```

Comment:
The message in InvalidTestCase is only used in one of the new test case. Do we really need to add this new field?
Ideally, the test case example code should explain itself clearly. The example code can also have some comments to explain it better. I'd suggest not to add InvalidTestCase.message to keep it simpler.

Reply:
I wanted to test the custom messages are working, not only show they exist. This allows testing custom messages and it doesn't add much complexity, only a new field that can be tested. we could call the field "`expected_message`" though. What do you think?

Reply:
Yeah, `expected_message` is a better name. I can see it can be useful for rules provide custom messages.